python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -317,12 +317,18 @@ def build_caching_repository_data_from_list(
             )
 
     assets_without_keys = [ad for ad in assets_defs if not ad.keys]
+
+    # Ensure that the same AssetsDefinition doesn't need to be considered twice by the graph if
+    # it produces multiple asset keys
+    deduped_asset_defs: list[Union[AssetsDefinition, SourceAsset]] = list(
+        {*assets_defs_by_key.values()}
```

Comment:
nit: could just do `*set(assets_defs_by_key.values())` in the list comprehension below